saagieapi/projects/graph_pipeline.py

```diff
@@ -0,0 +1,79 @@
+from typing import List
+
+import uuid
+
+
+class Node:
+    def __init__(self) -> None:
+        self.id = uuid.uuid4()
+
+
+class JobNode(Node):
+    def __init__(self, job_id) -> None:
+        super().__init__()
+        self.job_id = job_id
+        self.next_nodes: List[Node] = []
+
+    def add_next_node(self, node: Node):
+        self.next_nodes.append(node)
+
+
+class ConditionNode(Node):
+    def __init__(self) -> None:
+        super().__init__()
+        self.next_nodes_success: List[Node] = []
+        self.next_nodes_failure: List[Node] = []
+
+    def add_success_node(self, node: Node):
+        self.next_nodes_success.append(node)
+
+    def add_failure_node(self, node: Node):
+        self.next_nodes_failure.append(node)
+
+
+class GraphPipeline:
+    def __init__(self) -> None:
+        self.root_nodes = []
+        self.list_job_nodes = []
+        self.list_conditions_nodes = []
+
+    def add_root_node(self, node: JobNode):
+        self.root_nodes.append(node)
+
+    def fill_nodes_lists(self, node):
+        if str(node.id) not in [jn.get("id") for jn in self.list_job_nodes] and str(node.id) not in [cn.get("id") for cn in
+                                                                                                self.list_conditions_nodes]:
+
+            print(type(node))
+            if type(node) == JobNode:
+                print(f"job node: {node.id}")
```

Comment:
This print is used for debug? Remove it or replace with a logging.debug command instead if this is needed